File: src/main/java/br/com/rotaractsorocabaleste/rotasales/item/ItemController.java
package br.com.rotaractsorocabaleste.rotasales.item;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;
import java.util.Set;
import java.util.UUID;

@RestController
@RequestMapping("/items")
@RequiredArgsConstructor
public class ItemController {

    private final ItemService itemService;

    @GetMapping
    public Set<Item> findItemsByEventId(@RequestParam final UUID eventId) {
        return itemService.findItemsByEventId(eventId);
    }

    @PostMapping
    public Item create(@RequestBody @Valid final Item item) {
        return itemService.create(item);
    }

}

File: src/main/kotlin/br/com/rotaractsorocabaleste/rotasales/entrypoint/controller/ItemController.kt
package br.com.rotaractsorocabaleste.rotasales.entrypoint.controller

import br.com.rotaractsorocabaleste.rotasales.core.entity.Item
import br.com.rotaractsorocabaleste.rotasales.core.service.ItemService
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.net.URI
import java.util.*

@RestController
@RequestMapping("/items")
class ItemController(
    private val itemService: ItemService,
    private val logger: Logger = LoggerFactory.getLogger(ItemController::class.java)
) {

    @GetMapping
    fun getAllByEvent(
        @RequestParam("event") eventId: UUID
    ): ResponseEntity<List<Item>> {
        val ret = itemService.findByEventId(eventId)

        logger.info("Getting items for eventId=${eventId}")

        return ResponseEntity.ok(ret)
    }

    @PostMapping
    fun create(@RequestBody item: Item): ResponseEntity<Item> {
        logger.info("Received request for create an item, item description=${item.description}")

        val ret = itemService.save(item)

        logger.info("New item saved, item description=${ret.description}")

        return ResponseEntity.created(URI("/items")).build()
    }

}